ArgumentResolver/EntityValueResolver.php

```diff
@@ -0,0 +1,293 @@
+            $message = sprintf('"%s" object not found by the "%s" Argument Resolver.', $class, self::class);
+            if ($errorMessage) {
+                $message .= ' ' . $errorMessage;
+            }
+
+            throw new NotFoundHttpException($message);
+        }
+
+        return [$object];
+    }
+
+    private function getManager(?string $name, ?string $class): ?ObjectManager
+    {
+        if ($name === null) {
+            return $this->registry->getManagerForClass($class);
+        }
+
+        return $this->registry->getManager($name);
+    }
+
+    /**
+     * @param array<string, string> $options
+     *
+     * @return false|object|null
+     */
+    private function find(string $class, Request $request, array $options, string $name)
+    {
+        if ($options['mapping'] || $options['exclude']) {
+            return false;
+        }
+
+        $id = $this->getIdentifier($request, $options, $name);
+        if ($id === false || $id === null) {
+            return false;
+        }
+
+        $om = $this->getManager($options['entity_manager'], $class);
+        if ($options['evict_cache'] && $om instanceof EntityManagerInterface) {
+            $cacheProvider = $om->getCache();
+            if ($cacheProvider && $cacheProvider->containsEntity($class, $id)) {
+                $cacheProvider->evictEntity($class, $id);
+            }
+        }
+
+        try {
+            return $om->getRepository($class)->find($id);
+        } catch (NoResultException | ConversionException $e) {
+            return null;
+        }
+    }
+
+    /**
+     * @param array<string, string> $options
+     *
+     * @return false|mixed|mixed[]
+     */
+    private function getIdentifier(Request $request, array $options, string $name)
+    {
+        if ($options['id'] !== null) {
+            if (is_array($options['id'])) {
+                $id = [];
+                foreach ($options['id'] as $field) {
+                    // Convert "%s_uuid" to "foobar_uuid"
+                    if (strstr($field, '%s') !== false) {
+                        $field = sprintf($field, $name);
+                    }
+
+                    $id[$field] = $request->attributes->get($field);
+                }
+
+                return $id;
+            }
+
+            $name = $options['id'];
+        }
+
+        if ($request->attributes->has($name)) {
+            return $request->attributes->get($name);
+        }
+
+        if ($request->attributes->has('id') && ! $options['id']) {
+            return $request->attributes->get('id');
+        }
+
+        return false;
+    }
+
+    /**
+     * @param array<string, string> $options
+     *
+     * @return false|object|null
+     */
+    private function findOneBy(string $class, Request $request, array $options)
+    {
+        if (! $options['mapping']) {
+            $keys               = $request->attributes->keys();
+            $options['mapping'] = $keys ? array_combine($keys, $keys) : [];
```

Comment:
Making this configurable globally would means bundles won't be able to rely on one the the behaviors. My take on this would be to completely remove the feature as something that is to much magic for an argument resolver.

Reply:
well, I'm also fine with removing the magic mapping entirely too (the `mapping` option should still exist for cases where it is configured explicitly, as it solves a use case)

Reply:
> use any other request param 

other than what? `id` and the attribute with the same name than the argument will both use the method `find` . here we are talking about the fallback that use attributes that match none of the 2 cases above.

Disabling this feature globally does not make sens. If the request does not contain `id` nor an attribute with the same, then there will be nothing to match, entity could not be found, and user would have to provide a mapping on every attributes anyway.

Sorry, I disagree with your 

`/blogs/{slug}` or `/product/{uuid}` just work, and is often used. No need for custom mapping in such cases.

At the opposite, I never had issue with route containing several attributes. maybe I'm lucky, or because routes with multiple arguments are rare enough to provide mapping or expr and are not ambiguous

Reply:
anyway, I added a configuration section and added an `auto_mapping` flag that allows users to disable this behavior. This will enforce people to either use `{id}` (or the same name as the argument) OR to explicitly provide a `mapping` option (or `id` or `expr`).